src/main/java/bio/terra/janitor/service/iam/IamService.java

```diff
@@ -0,0 +1,33 @@
+package bio.terra.janitor.service.iam;
+
+import bio.terra.janitor.app.configuration.IamConfiguration;
+import bio.terra.janitor.common.exception.UnauthorizedException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/**
+ * IAM Service which handles authorization. Currently, Janitor uses a simple user email allow list
+ * to check user permission.
```

Comment:
Please add somewhere in this class-level comment (and maybe somewhere in the README too) some information about how service-level auth works here. My understanding (I had to go back & read the Janitor auth doc to refresh my memory) is that all Janitor clients will make Janitor calls with an access token from a specific service account, and this service account will be included in the allowlist?

Reply:
Actually there is no service authZ for admin endpoints use cases. Those 'admin endpoints are for users'(like admin users run things locally or via scripts). We don't have any use cases in which some services want to access admin endpoints using their service token. 
All services just use the 'specific service account' to publish messages, and service authZ is controlled by pubsub IAM policy. 
So I think the allowlist would just contains user emails. And even if any services in the future want to use admin endpoints, it should trigged by a user,  admin user token is expected in the request header, not service token.

Reply:
Aha, thanks – I had forgotten that we'd decided that service clients won't use HTTP directly.